data/tutorials/lg_08_error_handling.md

```diff
@@ -196,6 +766,25 @@ match Sys.os_type with
 | _ -> failwith "this system is not supported"
 ```
 
-Note a couple of extra pattern matching features in this example too. A
-so-called "range pattern" is used to match either `"Unix"` or
-`"Cygwin"`, and the special `_` pattern which matches "anything else".
+It is right to use `failwith` because using `assert` would be incorrect. Here is
+the dual example:
+```ocaml
+function x when true -> () | _ -> assert false
+```
+Here, it wouldn't be beneficial to use `failwith` since it requires the compiler
+to be bugged or the system to be corrupted for second code path to be executed.
+Breakage of the language semantics qualifies as extraordinary circumstances; it
+is catastrophic.
+
+# Concluding Remarks
+
+Properly handling errors is a complex matter. It is [cross-cutting
+concern](https://en.wikipedia.org/wiki/Cross-cutting_concern), touches all parts
+of an application and can't be isolated in a dedicated module. In contrast to
+several other mainstream languages, OCaml provides several mechanisms to handle
+exceptional circumstances, all with good runtime performance and code
+understandability. Using them properly requires some initial learning and
+practice. Later, it always requires some thinking, which is good since proper
+error management shouldn't ever be overlooked. No error handling is better
+than the others, and is should be matter of adequacy to the context rather than
+of taste. But opiniated OCaml code is also fine, so it's a balance.
```

Comment:
```suggestion
Properly handling errors is a complex matter. It is [cross-cutting
concern](https://en.wikipedia.org/wiki/Cross-cutting_concern), touches all parts
of an application, and can't be isolated in a dedicated module. In contrast to
several other mainstream languages, OCaml provides several mechanisms to handle
exceptional circumstances, all with good runtime performance and code
understandability. Using them properly requires some initial learning and
practice. Later, it always requires some thinking, which is good since proper
error management shouldn't ever be overlooked. No error handling is better
than the others, and is should be matter of adequacy to the context rather than
of taste. But opinionated OCaml code is also fine, so it's a balance.
```